daq/runner.py

```diff
@@ -35,12 +35,15 @@ class owns the main event loop and shards out work to subclasses."""
     _RESULT_LOG_FILE = 'inst/result.log'
 
     def __init__(self, config):
+        # pylint: disable=too-many-statements
         self.config = config
         self.port_targets = {}
         self.port_gateways = {}
         self.mac_targets = {}
         self.result_sets = {}
-        self._active_ports = {}
+        self._active_ports = set()
+        self._mac_port_mapping = {}
+        self._toggling_ports = {}
```

Comment:
I think we should move away form multiple top-level things that have the same key.  E.g., I think just one map (not set) for _active_ports would work, and then the "toggling ports" timeout information could be a field inside of that.  In the long run, it makes management of state a lot easier because when you, e.g. delete an active port, you don't have to manage the state in so many different places.